managed/src/main/java/com/yugabyte/yw/common/CertificateHelper.java

```diff
@@ -275,12 +281,35 @@ public static UUID uploadRootCA(
     }
     UUID rootCA_UUID = UUID.randomUUID();
     String keyPath = null;
-    if (certType == CertificateInfo.Type.SelfSigned) {
-      keyPath = String.format("%s/certs/%s/%s/ca.key.pem", storagePath,
-                                   customerUUID.toString(), rootCA_UUID.toString());
+    InputStream in = null;
+    try {
+      X509Certificate cert = null;
+      byte[] certEntryBytes = certContent.getBytes();
+      in = new ByteArrayInputStream(certEntryBytes);
+      CertificateFactory certFactory = CertificateFactory.getInstance("X.509");
+      cert = (X509Certificate) certFactory.generateCertificate(in);
+      if (certType == CertificateInfo.Type.SelfSigned) {
+        boolean isValid = verifySignature(cert, keyContent);
+        if (!isValid) {
+          throw new RuntimeException("Invalid certificate or Key.");
+        }
+        keyPath = String.format("%s/certs/%s/%s/ca.key.pem", storagePath,
+            customerUUID.toString(), rootCA_UUID.toString());
+      } else {
+        boolean isValidCert = isValidCACert(cert);
+        if (!isValidCert) {
+          throw new RuntimeException("Invalid certificate.");
+        }
+      }
+    } catch (Exception e) {
+      throw new RuntimeException("Invalid certificate.");
+    } finally {
+      if (in != null) {
```

Comment:
Use this: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html